### PR TITLE
add proto folder change

### PR DESCRIPTION
**Pull request by nghuyenthevinh2000**
Add change to proto folder and relevant scripts

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

**Comment by asalzmann** [comment]
Thank you for splitting this out! I'll test this out today

File: proto/stride/stakeibc/callbacks.proto [hunk 1]
```diff
@@ -11,50 +11,37 @@ message SplitDelegation {
...
 
 // ---------------------- Redemption Callbacks ---------------------- //
 message RedemptionCallback {
-  string hostZoneId = 1; 
-  repeated uint64 epochUnbondingRecordIds = 2;
-}
-
-// ---------------------- Validator Rebalance Callbacks ---------------------- //
-
-message Rebalancing {
```
Comment (sampocs) [inline proto/stride/stakeibc/callbacks.proto:51]: why is this removed?

File: proto/stride/stakeibc/params.proto [hunk 2]
```diff
@@ -32,5 +32,4 @@ message Params {
   uint64 safety_min_redemption_rate_threshold = 14;
   uint64 safety_max_redemption_rate_threshold = 15;
   uint64 ibc_transfer_timeout_nanos = 16;
-  uint64 safety_num_validators = 17;
```
Comment (sampocs) [inline proto/stride/stakeibc/params.proto:35]: I don't think this should be removed (same with the `next id` above)

**Review by sampocs — COMMENTED** [review]
Thanks for splitting this out, this looks great!

Looks like there were a couple proto changes that I think could probably be attributed to your branch being slightly out of date. Have you rebased with main recently?

File: proto/stride/stakeibc/callbacks.proto (hunk 1 above)
Comment (nghuyenthevinh2000) [reply proto/stride/stakeibc/callbacks.proto:51]: right, our notional-labs/stride is out of date

File: Makefile [hunk 3]
```diff
@@ -4,32 +4,34 @@ BUILDDIR ?= $(CURDIR)/build
...
-    else
-      GCC = $(shell command -v gcc 2> /dev/null)
-      ifeq ($(GCC),)
-        $(error gcc not installed for ledger support, please install or set LEDGER_ENABLED=false)
-      else
-        build_tags += ledger
-      endif
-    endif
-  endif
+	ifeq ($(OS),Windows_NT)
```
[the remark this answers is not in the document]
Comment (nghuyenthevinh2000) [reply Makefile:15]: yep, I use a script to format Makefile. Will change it back.

**Comment by nghuyenthevinh2000** [comment]
> After rebasing, I think this will be in good shape!

That should be it, I have double checked with main for any changes within 1 month to make sure.

File: proto/stride/stakeibc/params.proto [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-package Stridelabs.stride.stakeibc;
+package stride.stakeibc;
 
 import "gogoproto/gogo.proto";
 
 option go_package = "github.com/Stride-Labs/stride/x/stakeibc/types";
 
 // Params defines the parameters for the module.
-// next id: 18
+// next id: 16
```
Comment (sampocs) [inline proto/stride/stakeibc/params.proto:9]: I think we still want this to be 18?

**Review by sampocs — COMMENTED** [review]
Looks good! One small comment below

**Comment by sampocs** [comment]
@nghuyenthevinh2000 can you confirm the proto-gen makefile command still works for you after this rebase?
I just tested it and it errored for me, but it's possible it's a mac/linux thing

**Comment by nghuyenthevinh2000** [comment]
> @nghuyenthevinh2000 can you confirm the proto-gen makefile command still works for you after this rebase? I just tested it and it errored for me, but it's possible it's a mac/linux thing

all good

**Comment by sampocs** [comment]
@asalzmann can you see if the protogen works for you?

**Comment by sampocs** [comment]
Actually, nevermind this works now after the new commits!

**Review by sampocs — APPROVED** [review]
